### DISCUSSION TITLE
Как увеличить время жизни сессии?

### DISCUSSION THREAD

**Discussion by erpic-ru** [totumonline/totum-ru-issues-and-discussions#135] [Как это сделать?] [answered]
Подскажите, пожалуйста, как его изменить?

**Comment by Qb1cc** [comment]
Вот да, время сессии тоже интересует как изменить.

**Comment by ttm-alexeypolunin** [comment, marked as answer]
Итак, параметр `PHP` — `session.gc_maxlifetime`, измеряется в секундах. Те 4 часа = `14400`.

### Если у вас нативная установка по нашей инструкции:

Открываем настройки пула `FPM`:

```
nano /etc/php/8.0/fpm/pool.d/totum.conf
```

Если не по нашей инструкции, то найдите где у вас лежит файл пула.

И изменяем параметр `session.gc_maxlifetime`:

```
php_admin_value[session.gc_maxlifetime] = 14400
```

**Если у вас старая установка и нет такой строчки, то добавьте:**

    php_admin_value[session.gc_maxlifetime] = 14400
    php_admin_value[session.save_path] = /var/lib/php/sessions_totum
    

    и создайте папку для хранения сессий:

    
    mkdir /var/lib/php/sessions_totum && chown -R root:root /var/lib/php/sessions_totum && chmod 1733 /var/lib/php/sessions_totum
    

Рестартуем `FPM`:

```
service php8.0-fpm restart
```

### Если у вас Docker по нашей автоустановке (только с версии 3.4.43!):

Открываем настройки пула `FPM`:

```
sudo nano /home/totum/totum-mit-docker/nginx_fpm_conf/totum_fpm.conf
```

Если не по нашей инструкции, то найдите где у вас лежит файл пула.

И изменяем параметр `session.gc_maxlifetime`:

```
php_admin_value[session.gc_maxlifetime] = 14400
```

Рестартуем Docker:

```
cd /home/totum/totum-mit-docker/ && sudo docker-compose up --force-recreate -d
```

### Если у вас Netangels:

1. Ввести `session.gc_maxlifetime = 14400`
2. Добавить
3. Перезапустить `PHP`

<img width="1016" alt="Снимок экрана 2022-05-20 в 15 53 26" src="https://user-images.githubusercontent.com/55755565/169534564-54e8195b-5570-4359-9b37-7388fb97b79f.png">

**Reply by Alexmeb** [reply]
И еще обязательно для NetAngels в самом вверху нажать "Сохранить настройки"

**Comment by juramebel** [comment]
Почему то на нетангеле эта настройка спустя какое то время пропадает...

**Reply by ttm-alexeypolunin** [reply]
Это их панель, мы на нее никак не влияем. Можем предложить переехать в наше облако — те мы развернем и будем админить виртуалку с вашим проектом (даже на самом простом железе будет работать лучше, чем на NA-шареде):

На текущий момент цены такие:

### Цены на облако:

#### CPU ~ 2.2 GHz
T1 (2core, 1gb, 10gb SSD) — **3 239 р./мес**
T2 (2core, 2gb, 10gb SSD) — **3 902 р./мес**
T3 (4core, 4gb, 20gb SSD) — **5 450 р./мес**

`404 р./мес за каждые дополнительные 10 гб на SSD`

#### CPU ~ 4.8 GHz
TTM-F1 (2core, 1gb, 10gb NVMe) — **5 030 р./мес**
TTM-F2 (2core, 2gb, 10gb NVMe) — **6 680 р./мес**
TTM-F3  (4core, 4gb, 20gb NVMe) — **11 450 р./мес**

`536 р./мес за каждые дополнительные 10 GB на NVMe`

+ Один полный бэкап сервера в день включен в стоимость.
+ Домен на хосте `{youdomain}.ttmapp.ru`
+ Парковка вашего домена + 500 р./мес
+ 1 схема на сервере + 1 000 р./мес за каждую дополнительную схему.
+ Дополнительные бекапы схемы 3 раза в день для экстренного восстановления + 1 000 р./мес за схему (последние 7 дней, 10 GB на бекапы).
+ Полное восстановление бекапа — 3 000 р.
+ Частичное восстановление бекапа — 3 000 р. + по часам 3 000 р./час
+ Обновление Totum по запросу — 2 000 р. минорная версия. / 6 000 р. мажорная версия (при оплате 6 мес сразу — минорные бесплатно, при оплате года вперед — мажорные бесплатно, но не более 3-х схем на одном сервере).
